sycamore/transforms/merge_elements.py

```diff
@@ -0,0 +1,130 @@
+from abc import ABC, abstractmethod
+
+from ray.data import Dataset
+
+from sycamore.data import Document, Element, BoundingBox
+from sycamore.plan_nodes import NonCPUUser, NonGPUUser, Transform, Node
+from sycamore.transforms.map import generate_map_function
+
+from typing import Tuple
+
+
+
+class ElementMerger(ABC):
+    @abstractmethod
+    def merge_elements(self, document: Document) -> Document:
+        pass
+
+class GreedyElementMerger(ElementMerger):
```

Comment:
Should be explicit this is a text based merger, we'll want other types of mergers that will also be using the greedy approach (e.g. merge overlapping boxes until text is full)

Reply:
ack. `GreedyTextElementMerger`?